### PR TITLE
gh-113081: Print colorized exception just like built-in traceback in pdb

### PR DESCRIPTION
**Pull request by gaogaotiantian**
This is the first step to colorize pdb - just use the existing colorized traceback for exceptions in post mortem mode. The code change is really trivial and it just calls the colorized version of exception printing in traceback.

Yes, it's using an internal function, but `pdb` is stdlib so it can adapt when the function is changed. We can also use `print_exception` but we need `traceback._can_colorize()` anyway - which should be put into a separate module if we want to colorize more modules.

* Issue: gh-113081

### CODE REVIEW

**Comment by gaogaotiantian** [comment]
@pablogsal could you share some thoughts on this? Do you think we need extra tests for pdb? Thanks!

File: Lib/pdb.py [hunk 1]
```diff
@@ -2232,7 +2232,7 @@ def main():
             print("The program exited via sys.exit(). Exit status:", end=' ')
             print(e)
         except BaseException as e:
-            traceback.print_exc()
+            traceback._print_exception_bltin(e)
```
Comment (brandtbucher) [inline Lib/pdb.py]: > We can also use `print_exception` but we need `traceback._can_colorize()` anyway - which should be put into a separate module if we want to colorize more modules.

`_colorize` is now its own module. Let's change this:

```suggestion
            traceback._print_exception_bltin(e, colorize=_colorize.can_colorize())
```

**Comment by brandtbucher** [comment]
@pablogsal, any thoughts on this simple (colorful!) change? Otherwise I'm inclined to merge. :)

File: Lib/pdb.py (hunk 1 above)
Comment (gaogaotiantian) [reply Lib/pdb.py]: Just double checked, `_print_exception_bltin` only takes an exception. It queries `_colorize` inside the function.

Comment (gaogaotiantian) [reply Lib/pdb.py]: Oh you want me to switch to `print_exception` the public interface?

**Review by brandtbucher — APPROVED** [review]

**Comment by gaogaotiantian** [comment]
What is the current proper way to disable color in tests? I thought there was a global mechanism implemented. I can do it for pdb only but want to make sure this is not something that should be covered already.

**Comment by brandtbucher** [comment]
Check out `test.support.force_not_colorized`.

**Comment by brandtbucher** [comment]
Looks like `test_issue84583` needs a decorator too.

**Comment by gaogaotiantian** [comment]
Okay seems like the tests passed now. My local tests worked fine and I did not have the time to figure out the difference.

**Review by pablogsal — APPROVED** [review]

**Comment by pablogsal** [comment]
Let's merge it!

**Pull request merged**